spec/requests/api/v1/serializer_options_spec.rb

```diff
@@ -0,0 +1,53 @@
+require 'cgi'
+require 'uri'
+
+describe 'Serializer options', type: :request do
+  it 'adds meta with pagination information' do # rubocop:disable RSpec/MultipleExpectations
+    get '/api/v1/locations', params: {}, headers: default_headers
+
+    expect(response).to have_http_status(:ok)
+    expect(json_response).to have_key('meta')
+    meta = json_response['meta']
+    expect(meta).to have_key('current_page')
+    expect(meta).to have_key('total_pages')
+    expect(meta).to have_key('total_count')
+    expect(meta).to have_key('padding')
+    expect(meta).to have_key('page_size')
+    expect(meta).to have_key('max_page_size')
+  end
+
+  it 'adds links' do
+    get '/api/v1/locations', params: {}, headers: default_headers
+
+    expect(response).to have_http_status(:ok)
+    expect(json_response).to have_key('links')
+    links = json_response['links']
+    expect(links).to have_key('self')
+    expect(links).to have_key('first')
+    expect(links).to have_key('last')
+  end
+
+  it 'adds client requested fields information to links' do
+    q = { fields: { locations: 'latitude' } }
+    get "/api/v1/locations?#{q.to_query}", params: {}, headers: default_headers
+
+    expect(response).to have_http_status(:ok)
+    self_link = json_response.dig('links', 'self')
+    expect(parse_query(self_link)).to have_key('fields[locations]')
```

Comment:
This is a bit hard to parse, could we do a regex compare instead? It seems like string comparison at a glance 🤔 Same comment for the assertion below

Reply:
In this case, regex compare wouldn't work out-of-the-box because `[`, `]` are URL encoded.

I've played around with method naming and came up with following
```ruby
    expect(query_params(json_response.dig('links', 'self')))
      .to include('fields[locations]' => ['latitude'])
```

What do you think?

Reply:
I had something like this in mind

```ruby
parsed_query = parse_query json_response.dig('links', 'self')
expect(parsed_query).to have_key('fields[locations]')
```

Reply:
If I rate these in the order of mental capacity required to parse them (descending order) I come up with the following

```ruby
self_link = json_response.dig('links', 'self')
expect(parse_query(self_link)).to have_key('fields[locations]')
```

> what is parse query, what is self link

```ruby
  expect(query_params(json_response.dig('links', 'self')))
      .to include('fields[locations]' => ['latitude'])
```
> what is query params, why is the return value a hash

```ruby
expect(json_response.dig('links', 'self')).to match('lectures?fields%5Bself%5D=something')
```

> checking JSON response element against a string regex and bob's your uncle

Option 2 seems like a good middle ground but I will defer the decision to you since this is a nitpick 👍